### PR TITLE
Add superset test for bloomfilters

### PR DESCRIPTION
**Pull request by krl**
This should be the last thing i need for now...

### CODE REVIEW

File: blocks/bloom/filter.go [hunk 1]
```diff
@@ -15,6 +15,7 @@ type Filter interface {
 	Find([]byte) bool
 	Merge(Filter) (Filter, error)
 	HammingDistance(Filter) (int, error)
+	SupersetOf(Filter) (bool, error)
```
Comment (jbenet) [inline blocks/bloom/filter.go]: I'm thinking this is the wrong way to add these extensions-- they're putting everything into a single interface. I think maybe the `container/heap` -- http://golang.org/pkg/container/heap/ -- approach is the right one.

So: 
```go
func Superset(a, b Filter) (bool, error) {
  // return whether a is superset of b
}
```

Comment (jbenet) [inline blocks/bloom/filter.go]: the `strings` package is similar: http://golang.org/pkg/strings/

**Comment by jbenet** [comment]
@krl still want this in, or no? Has been siting here for a while. If we want it in, let's reopen and fix it

**Comment by krl** [comment]
Juan Batiz-Benet <notifications@github.com> writes:

> Closed #1100.
>
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/ipfs/go-ipfs/pull/1100#event-344533755

No worries, i will probably rewrite anything that depends on this stuff.

**Pull request closed without merging**